src/main/java/com/digitalasset/testing/junit4/Sandbox.java

```diff
@@ -110,19 +115,35 @@ public SandboxBuilder logLevel(LogLevel logLevel) {
       return this;
     }
 
+    public SandboxBuilder projectRoot(Path projectRoot) {
+      try {
+        if (Files.list(projectRoot).noneMatch(damlYamlP()))
+          throw new IllegalArgumentException("Project root must contain a daml.yaml");
+
+        this.projectRoot = projectRoot;
+        return this;
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+
     public Sandbox build() {
       Objects.requireNonNull(darPath);
+      if (projectRoot == null) {
+        projectRoot = findDamlYaml(darPath.toAbsolutePath().getParent());
```

Comment:
```suggestion
        projectRoot = darPath.toAbsolutePath().getParent();
```
because the assistant will `findDamlYaml`